Examples/ObjectiveC/ManageOfflineRegionsExample.m

```diff
@@ -3,12 +3,281 @@
+                                             frame.width / 2, 10);
+        [self.view addSubview:self.progressView];
+    }
+
+    [self.progressView setProgress:progressPercentage animated:YES];
+
+    // If this pack has finished, print its size and resource count.
+    if (completedResources == expectedResources) {
+        NSString *byteCount = [NSByteCountFormatter
+                               stringFromByteCount:progress.countOfBytesCompleted countStyle:NSByteCountFormatterCountStyleMemory];
+        NSLog(@"Offline pack “%@” completed: %@, %llu resources",
+              userInfo[@"name"], byteCount, completedResources);
+    } else {
+        // Otherwise, print download/verification progress.
+        NSLog(@"Offline pack “%@” has %llu of %llu resources — %.2f%%.",
+              userInfo[@"name"], completedResources,
+              expectedResources, progressPercentage * 100);
+    }
+
+    // Reload the table to update the progress percentage for each offline pack.
+    [self.tableView reloadData];
+
+}
+
+#pragma mark - UITableView configuration
+
+// Create the table view which will display the downloaded regions.
+- (NSInteger)tableView:(nonnull UITableView *)tableView
+ numberOfRowsInSection:(NSInteger)section {
+    if (MGLOfflineStorage.sharedOfflineStorage.packs.count != 0) {
+        return MGLOfflineStorage.sharedOfflineStorage.packs.count;
+    } else {
+        return 0;
+    }
+
+}
+
+- (UIView *)tableView:(UITableView *)tableView
+viewForHeaderInSection:(NSInteger)section {
+    UILabel * label = [[UILabel alloc] init];
+    label.backgroundColor = UIColor.systemBlueColor;
+    label.textColor = UIColor.whiteColor;
+    label.font = [UIFont preferredFontForTextStyle: UIFontTextStyleHeadline];
+    label.textAlignment = NSTextAlignmentCenter;
+
+
+    if (MGLOfflineStorage.sharedOfflineStorage.packs != nil) {
+        label.text = @"Offline maps";
+    } else {
+        label.text = @"No offline maps";
+    }
+
+    return label;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView
+heightForHeaderInSection:(NSInteger)section {
+    return 50.0;
+}
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView
+                 cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
+
+    [cell initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:@"cell"];
+    
+    if (MGLOfflineStorage.sharedOfflineStorage.packs.count != 0 )  {
+        MGLOfflinePack * pack = MGLOfflineStorage.sharedOfflineStorage.packs[indexPath.row];
+        long number = indexPath.row + 1;
+        long countOfBytesCompleted = MGLOfflineStorage.sharedOfflineStorage.packs[indexPath.row].progress.countOfBytesCompleted;
+        NSByteCountFormatter *formattedCountOfBytedCompleted = [NSByteCountFormatter stringFromByteCount:countOfBytesCompleted countStyle:NSByteCountFormatterCountStyleMemory];
```

Comment:
Here I'm seeing a warning:

```
Incompatible pointer types initializing 'NSByteCountFormatter *' 
with an expression of type 'NSString * _Nonnull'
```

This is because the method `[NSByteCountFormatter stringFromByteCount:countStyle:]` returns `NSString`, and not `NSByteCountFormatter` which is defined before the variable name.

Reply:
🤦🏾‍♀️, thank you!